src/utils/utilities.jl

```diff
@@ -210,6 +210,48 @@ function prettytime(t)
     return @sprintf("%.3f %s", value, units)
 end
 
+import Dates
+using Dates
+
+time_per_time(::Type{P}, ::Type{P}) where {P} = 1
+function define_time_per_times(periods)
+    for i in eachindex(periods)
+        T, n = periods[i]
+        N = Int64(1)
+        for j in (i - 1):-1:firstindex(periods) # less-precise periods
+            Tc, nc = periods[j]
+            N *= nc
+            @eval time_per_time(::Type{$T}, ::Type{$Tc}) = $N
+        end
+    end
+end
```

Comment:
Can you please add a comment to explain this?

Reply:
Sure, I'll add some docs (I took this from Dates, but it wasn't really documented there). Ultimately, this is just defining:

```julia
time_per_time(::Type{Dates.Second}, ::Type{Dates.Minute}) = 60
time_per_time(::Type{Dates.Second}, ::Type{Dates.Hour}) = 3600
time_per_time(::Type{Dates.Second}, ::Type{Dates.Day}) = 86400
time_per_time(::Type{Dates.Second}, ::Type{Dates.Week}) = 604800
time_per_time(::Type{Dates.Millisecond}, ::Type{Dates.Second}) = 1000
time_per_time(::Type{Dates.Millisecond}, ::Type{Dates.Minute}) = 60000
time_per_time(::Type{Dates.Millisecond}, ::Type{Dates.Hour}) = 3600000
time_per_time(::Type{Dates.Millisecond}, ::Type{Dates.Day}) = 86400000
time_per_time(::Type{Dates.Millisecond}, ::Type{Dates.Week}) = 604800000
time_per_time(::Type{Dates.Day}, ::Type{Dates.Week}) = 7
time_per_time(::Type{Dates.Hour}, ::Type{Dates.Day}) = 24
time_per_time(::Type{Dates.Hour}, ::Type{Dates.Week}) = 168
time_per_time(::Type{Dates.Minute}, ::Type{Dates.Hour}) = 60
time_per_time(::Type{Dates.Minute}, ::Type{Dates.Day}) = 1440
time_per_time(::Type{Dates.Minute}, ::Type{Dates.Week}) = 10080
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Microsecond}) = 1000
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Millisecond}) = 1000000
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Second}) = 1000000000
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Minute}) = 60000000000
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Hour}) = 3600000000000
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Day}) = 86400000000000
time_per_time(::Type{Dates.Nanosecond}, ::Type{Dates.Week}) = 604800000000000
time_per_time(::Type{Dates.Microsecond}, ::Type{Dates.Millisecond}) = 1000
time_per_time(::Type{Dates.Microsecond}, ::Type{Dates.Second}) = 1000000
time_per_time(::Type{Dates.Microsecond}, ::Type{Dates.Minute}) = 60000000
time_per_time(::Type{Dates.Microsecond}, ::Type{Dates.Hour}) = 3600000000
time_per_time(::Type{Dates.Microsecond}, ::Type{Dates.Day}) = 86400000000
time_per_time(::Type{Dates.Microsecond}, ::Type{Dates.Week}) = 604800000000
```